CMakeLists.txt

```diff
@@ -156,10 +156,35 @@ set(GMOCK_LIBS
 set(CMAKE_CXX_STANDARD 14)
 
 # Common configuration for all build modes.
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
 
+# does the compiler support std::atomic_load(&std::shared_ptr<T>) ?
+try_compile(HAS_NATIVE_ATOMIC_SHARED_PTR ${CMAKE_BINARY_DIR}/has_shared_ptr_support_rs
+    ${CMAKE_CURRENT_SOURCE_DIR}/yarpl/test/test_has_shared_ptr_support.cpp)
```

Comment:
You could use https://cmake.org/cmake/help/v3.0/module/CheckCXXSourceCompiles.html to write the C++ code inline here if you want.  I wouldn't block the change on it though.